### DISCUSSION TITLE
Is is possible to control what relationships to use?

### DISCUSSION THREAD

**Discussion by NaumenkoSergiy** [okuramasafumi/alba#325] [Q&A] [answered]
I'm interesting if it's possible to control relationships in next way:

```
App::UserSerializer.new(
  @users,
  include: [:contacts, :organization]
 )
```

I know it's possible with params, but would be good to have some approach like showed above.

**Comment by okuramasafumi** [comment]
@NaumenkoSergiy Could you provide the expected output?

**Comment by NaumenkoSergiy** [comment]
@okuramasafumi 

```
App::UserSerializer.new(
  @users,
  include: [:organization]
)
```
 =>
```
[
  {
    id: 1,
    full_name: 'Test Name',
    organization: {
      id: 1,
      ........
    }
  }
]
```

```
App::UserSerializer.new(
  @users
)
```
 =>
```
[
  {
    id: 1,
    full_name: 'Test Name'
  }
]
```

so, just control relationships with something like `include` keyword, or is it possible only with params?

**Reply by okuramasafumi** [reply, marked as answer]
https://github.com/okuramasafumi/alba#conditional-attributes
Does this help?
https://github.com/okuramasafumi/alba/blob/main/test/usecases/conditional_attributes_test.rb
This test case shows you how to use it, especially with `params`.

**Comment by NaumenkoSergiy** [comment]
@okuramasafumi thank you, it helps
